sapp/sarif.py

```diff
@@ -35,13 +47,13 @@
 ]
 
 
-@dataclass
 class SARIF:
     version: str = "2.1.0"
     schema: str = "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json"  # noqa
-    _tool_warning_code_ranges: Dict[str, Tuple[int, int]] = field(default_factory=dict)
-    driver: Dict[str, Union[str, List[Dict[str, str]]]] = field(default_factory=dict)
-    results: List[SARIFResult] = field(default_factory=list)
+    _tool_warning_code_ranges: Dict[str, Tuple[int, int]] = dict()
+    driver: Dict[str, Union[str, List[Dict[str, str]]]] = dict()
+    results: List[SARIFResult] = list()
```

Comment:
The problem is that those fields are initialized only once at the class level. This will lead to strange bugs:

```
class Foo:
  results = list()

Foo().results.append(1)
print(Foo().results) # shows [1] even though it's a different instance.
```

If we are not using `@dataclass` anymore we need to initialize all fields manually in the constructor.

Reply:
oh yes 🤦‍♂️